config/initializers/kafka.rb

```diff
@@ -48,3 +48,8 @@ def self.instance
       end
   end
 end
+
+# Instantiate the singleton here in bootstrapping.  This will prevent the
+# race condition of instantiation thru the request. 
```

Comment:
Can you add more of your PR comments here so they are preserved and more visible to future devs?

Reply:
done